url9e019a0ee9_elfrasco_design_principles_tgz-pJ8-ar_com_epidataconsulting_principles_ioc_before_SavePredictiveEngineProjectServiceJ8/src/ar/com/epidataconsulting/principles/ioc/before/SavePredictiveEngineProjectService.java

```diff
@@ -0,0 +1,123 @@
+import java.util.Calendar;
+import java.util.GregorianCalendar;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+
+import ar.com.epidataconsulting.principles.model.ProjectType;
+import ar.com.epidataconsulting.principles.model2.Analyst;
+import ar.com.epidataconsulting.principles.model2.Architect;
+import ar.com.epidataconsulting.principles.model2.Developer;
+import ar.com.epidataconsulting.principles.model2.Employee;
+import ar.com.epidataconsulting.principles.model2.Project;
+import ar.com.epidataconsulting.principles.model2.ProjectManager;
+import ar.com.epidataconsulting.principles.model2.Tester;
+
+public class SavePredictiveEngineProjectService {
+	
+	private static final Logger logger = Logger.getLogger(SavePredictiveEngineProjectService.class);
+
+	public void save() {
+		
+		// Create the Predictive Engine Project
+		Project project = new Project();
+		project.setName("Predictive Engine");
+		project.setClientName("OSDE");
+		project.setDurationInMonths(4);
+		project.setType(ProjectType.DEVELOPMENT);
+		project.setTeam(createTeam());
+		
+		// Persist the project
+		ProjectDAO projectDAO = new ProjectDAO();
+		projectDAO.save(project);
+		
+		// Recovery the project to assert
+		Project projectRecovered = projectDAO.findByName("Predictive Engine");
+		logger.debug("NAME = " + projectRecovered.getName());
+		logger.debug("CLIENT = " + projectRecovered.getClientName());
+		logger.debug("TEAM SIZE = " + projectRecovered.getTeam().size());
+		
+	}
+	
+	private List<Employee> createTeam() {
+		
+		Employee florencia = createFlorenciaCastagnino();		
+		Employee adrian = createAdrianParedes();		
+		Employee leonardo = createLeonardoPavanello();		
+		Employee juan = createJuanGaragorri();		
+		Employee veronica = createVeronicaPechersky();
+		Employee erica = createEricaQuispe();
+		
+		List<Employee> team = new ArrayList<Employee>();
+		team.add(florencia);
+		team.add(veronica);
+		team.add(adrian);
+		team.add(leonardo);
+		team.add(juan);
+		team.add(erica);
+		return team;
+	}
+
+	private Employee createFlorenciaCastagnino() {
+		Employee employee = new ProjectManager();
+		employee.setName("Florencia");
+		employee.setLastName("Castagnino");
+		employee.setDateOfBirth(new GregorianCalendar(1980, Calendar.JANUARY, 15).getTime());
+		employee.setFileNumber("123");
+		return employee;
+	}
+	
+	private Employee createAdrianParedes() {
+		Employee employee = new Architect();
+		employee.setName("Adrian");
+		employee.setSecondName("Marcelo");
+		employee.setLastName("Paredes");
+		employee.setDateOfBirth(new GregorianCalendar(1982, Calendar.JUNE, 3).getTime());
+		employee.setFileNumber("456");
+		return employee;
+	}
+	
+	private Employee createLeonardoPavanello() {
+		Employee employee = new Developer();
+		employee.setName("Leonardo");
+		employee.setLastName("Pavanello");
+		employee.setDateOfBirth(new GregorianCalendar(1984, Calendar.MAY, 28).getTime());
+		employee.setFileNumber("789");
+		return employee;		
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 84:89, 93:98.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.